File: app/src/main/java/com/jurtz/marcel/shoppinglist/model/ShoppingList.java
package com.jurtz.marcel.shoppinglist.model;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;
import java.util.List;

@Entity(tableName = "shopping_list")
public class ShoppingList {

    /*
    public ShoppingList(int id, String description, int timestampSeconds, List<ShoppingListItem> items) {
        this.id = id;
        this.description = description;
        this.timestampSeconds = timestampSeconds;
        this.items = items;
    }
    */

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    public int id;

    @ColumnInfo(name = "description")
    public String description;

    @ColumnInfo(name = "timestamp_seconds")
    public int timestampSeconds;

    @Ignore
    public List<ShoppingListItem> items;

    @Ignore
    public int itemCount;
}

File: app/src/main/java/com/jurtz/marcel/shoppinglist/model/ShoppingList.kt
package com.jurtz.marcel.shoppinglist.model

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Embedded
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

import java.util.Calendar

@Entity(tableName = "shopping_list")
data class ShoppingList(
        @ColumnInfo(name = "description")
        var description: String? = null,

        @ColumnInfo(name = "timestamp")
        var timestamp: Int = (Calendar.getInstance().timeInMillis / 1000).toInt(),

        @Embedded
        private var items : MutableList<ShoppingListItem> = arrayListOf()
) {

    @PrimaryKey(autoGenerate = true)
    private val id: Int = 0

    fun getItemCount() : Int {
        return items.size
    }

    fun addItem(item : ShoppingListItem) {
        items.add(item)
    }

    fun removeItem(item: ShoppingListItem) {
        items.remove(item)
    }
}
